### PR TITLE
Gltf materials

### PR DESCRIPTION
**Pull request by sebavan**
Add a way to either only download the gltf materials or to skip all of them.

### CODE REVIEW

**Review by deltakosh — APPROVED** [review]

File: loaders/src/glTF/glTFFileLoader.ts [hunk 1]
```diff
@@ -238,6 +238,16 @@ export class GLTFFileLoader implements IDisposable, ISceneLoaderPluginAsync, ISc
      */
     public loadAllMaterials = false;
 
+    /**
+     * If true, load only all the materials defined in the file. Defaults to false.
```
Comment (bghgary) [inline loaders/src/glTF/glTFFileLoader.ts:242]:
```suggestion
     * If true, load only the materials defined in the file. Defaults to false.
```

File: loaders/src/glTF/glTFFileLoader.ts [hunk 2]
```diff
@@ -238,6 +238,16 @@ export class GLTFFileLoader implements IDisposable, ISceneLoaderPluginAsync, ISc
...
 
+    /**
+     * If true, load only all the materials defined in the file. Defaults to false.
+     */
+    public loadOnlyMaterials = false;
+
+    /**
+     * If true, do not load any materials defined in the file. Defaults to false.
+     */
+     public skipMaterials = false;
```
Comment (bghgary) [inline loaders/src/glTF/glTFFileLoader.ts:249]:
```suggestion
    public skipMaterials = false;
```

**Review by bghgary — APPROVED** [review]
A couple of nits. I'm a bit concerned we are adding more and more options without doing to work to make setting options easier, but it should be okay for now.

File: loaders/src/glTF/glTFFileLoader.ts [hunk 3]
```diff
@@ -238,6 +238,22 @@ export class GLTFFileLoader implements IDisposable, ISceneLoaderPluginAsync, ISc
...
+    /**
+     * If true, do not load any materials defined in the file. Defaults to false.
+     */
+    public skipMaterials = false;
+
+    /**
+     * If defined, provides a callback to override a specific material from a GLTF file.
+     * If the callback returns null, the requested gltf material will be loaded.
+     */
+    public customMaterialLoad: Nullable<(material: GLTF2.IMaterial) => Nullable<Material>> = null;
```
Comment (bghgary) [inline loaders/src/glTF/glTFFileLoader.ts:255]: This is the same as adding a glTF loader extension?

**Review by bghgary — CHANGES_REQUESTED** [review]

**Pull request merged**